packages/web-client/app/resources/card-customization.ts

```diff
@@ -0,0 +1,102 @@
+import { timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { getResolver } from '@cardstack/did-resolver';
+import { Resolver } from 'did-resolver';
+import { Resource } from 'ember-resources';
+import config from '../config/environment';
+
+const FIRST_RETRY_DELAY = config.environment === 'test' ? 100 : 1000;
+
+interface Args {
+  named: {
+    customizationDID: string;
+    waitForCustomization: boolean;
+  };
+}
+
+export class CardCustomization extends Resource<Args> {
+  // our output values
+  @tracked issuerName: string | undefined;
+  @tracked background: string | undefined;
+  @tracked patternColor: string | undefined;
+  @tracked textColor: string | undefined;
+  @tracked patternUrl: string | undefined;
+
+  // our derived state so you can check when the output values are ready, if you
```

Comment:
Unsure about use of the word, is this derived state? It isn't calculated based on other state in here.

Reply:
I think it’s a reference to [similar properties](http://ember-concurrency.com/docs/derived-state) on Ember Concurrency tasks. But now that I think about it I think these comments were more to educate me than to be committed, so I’ll remove them 😆